enzy_htp/structure/structure_region/structure_region.py

```diff
@@ -325,50 +345,20 @@ def convert_to_structure(self, cap_as_residue = False) -> Structure:
             cap_as_residue: will not include the cap in the residue if True
         Returns:
             The corresponding Structure of the structure region"""
-        residue_mapper: {Residue, List[Atom]} = {}
-        res_to_cap_atoms: {Residue, int} = {}
 
-        for aa in self.atoms:
-            if isinstance(aa.parent, ResidueCap):
-                if cap_as_residue:
-                    continue
-                else:
-                    aa.residue = aa.parent.link_atom.residue
-                if aa.residue not in res_to_cap_atoms:
-                    res_to_cap_atoms[aa.residue] = 0
-                res_to_cap_atoms[aa.residue] += 1
-
-            if aa.residue not in residue_mapper:
-                residue_mapper[aa.residue] = []
-            residue_mapper[aa.residue].append(aa)
-
-        residues = []
-        for res, atoms in residue_mapper.items():
-            if res not in res_to_cap_atoms.keys():
-                res_to_cap_atoms[res] = 0
-            
-            if res.rtype == chem.ResidueType.MODIFIED:
-                new_res = ModifiedResidue(int(res.idx), res.name, atoms, res.parent, net_charge=int(res.net_charge), multiplicity=int(res.multiplicity))
-                new_res.renumber_atoms(range(1, len(atoms) + res_to_cap_atoms[res] + 1))
-                residues.append(new_res)
-            elif res.rtype == chem.ResidueType.LIGAND:
-                new_res = Ligand(int(res.idx), res.name, atoms, res.parent, net_charge=int(res.net_charge), multiplicity=int(res.multiplicity))
-                new_res.renumber_atoms(range(1, len(atoms) + res_to_cap_atoms[res] + 1))
-                residues.append(new_res)
-            elif res.rtype == chem.ResidueType.METAL:
-                new_res = MetalUnit(int(res.idx), res.name, atoms, res.parent)
-                new_res.renumber_atoms(range(1, len(atoms) + res_to_cap_atoms[res] + 1))
-                residues.append(new_res)            
-            else:
-                new_res = Residue(int(res.idx), res.name, atoms, res.parent)
-                new_res.renumber_atoms(range(1, len(atoms) + res_to_cap_atoms[res] + 1))
-                residues.append(new_res)
+        residue_mapper = self.involved_residue_atom_mapper(cap_as_residue=cap_as_residue)
+
+        residues: List[Residue] = []
+        for res in self.involved_residues:
+            new_res = deepcopy(res)
+            new_res.atoms = residue_mapper[res]
+            new_res.chain = res.chain
+            new_res.renumber_atoms(range(1, len(new_res.atoms) + 1))
+            residues.append(new_res)
```

Comment:
Instead of iterating the self.involved_residues, we would like to iterate the newly created residue_mapper.

Reply:
Seems caps are left out in this case. I will explain it better in the comment for `involved_residue_atom_mapper`

Reply:
In renumber_atoms, you can use `num_atoms` method instead of len()